docs/addons/secrets-store.md

```diff
@@ -0,0 +1,50 @@
+# Secrets Store Add-on
+
+The Secrets Store Add-on provisions the [AWS Secrets Manager and Config Provider for Secret Store CSI Driver(ASCP)](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver.html) on your EKS cluster. With ASCP, you now have a plugin for the industry-standard Kubernetes Secrets Store [Container Storage Interface (CSI) Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) used for providing secrets to applications operating on Amazon Elastic Kubernetes Service.
+
+With ASCP, you can securely store and manage your secrets in [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager) or [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) and retrieve them through your application workloads running on Kubernetes. You no longer have to write custom code for your applications.
+
+> **Compatibility**: This addon is only compatible with following Cluster Providers.<br/> [EC2 Cluster Provider](./../cluster-providers/ec2-cluster-provider.md)
+
+## Usage
+
+#### **`index.ts`**
+```typescript
+import { SecretsStoreAddOn, SecretType, ClusterAddOn, EksBlueprint }  from '@shapirov/cdk-eks-blueprint';
+
+const secretsStoreAddOn = new SecretsStoreAddOn({
+  rotationPollInterval: '120s',
+  secretsProviderConfigs: [
+    {
+      namespace: 'team-riker',
+      secrets: [
```

Comment:
Can we break this addon up so that the specific secrets for each team lives in the team implementation? And the Addon itself only does what is necessary to configure ASCP?

Reply:
addressed in the design